File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        ListNode nodeToBeDeleted = node;
        ListNode nextNode = nodeToBeDeleted.next; //it exists as node is not tail
        nodeToBeDeleted.val = nextNode.val;
        nodeToBeDeleted.next = nextNode.next;
        nextNode.next = null;
    }
}

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} node
 * @return {void} Do not return anything, modify node in-place instead.
 */
var deleteNode = function(node) {
    let prev = node, next = node.next;
    let tailprev = null;
    while(next) {
        let temp = prev.val;
        prev.val = next.val;
        next.val = temp;
        tailprev = prev;
        prev= next;
        next = next.next;
    }
    tailprev.next = null;
    
    
};